File: CommunicationLibrary/utils/StringUtils.cs
using System;

namespace CommunicationLibrary.utils {
    public static class StringUtils {
        public static bool GetRestOfString(string str, string start, string end, out string rest) {
            if(str.StartsWith(start) && str.EndsWith(end)){
				rest = str.Substring(start.Length, str.Length - end.Length - start.Length);
                return true;
            }else{
                rest = "";
                return false;
            }
        }

        public static bool GetRestOfStringSplited(string str, string start, string end, out string [] rest, params char[] separator) {
            string restStr;
            if (GetRestOfString(str, start, end, out restStr)) {
				rest = restStr.Split(separator, StringSplitOptions.RemoveEmptyEntries);
                return true;
            } else {
                rest = null;
                return false;
            }
        }

		public static bool GetRestOfStringSplited(string str, string start, string end, out string[] rest, params string[] separator) {
			string restStr;
			if (GetRestOfString(str, start, end, out restStr)) {
				rest = restStr.Split(separator, StringSplitOptions.RemoveEmptyEntries);
				return true;
			} else {
				rest = null;
				return false;
			}
		}
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/utils/StringUtils.java
package BaseLibrary.utils;

public final class StringUtils {
	private StringUtils() {
	}

	public static boolean GetRestOfString(String str, String start, String end, Holder<String> rest) {
		if (str.startsWith(start) && str.endsWith(end)) {
			rest.value = str.substring(start.length(), str.length() - end.length() - start.length());
			return true;
		} else {
			return false;
		}
	}

	public static boolean GetRestOfStringSplited(String str, String start, String end, Holder<String[]> rest,
			char... separator) {
		Holder<String> restStr = new Holder<>();

		if (GetRestOfString(str, start, end, restStr)) {
			rest.value = restStr.value.split(new String(separator));
			return true;
		} else {
			rest = null;
			return false;
		}
	}
}
